examples/VLCB_4in4out/VLCB_4in4out.ino

```diff
@@ -219,64 +223,89 @@ void loop()
   }
 
   // test for switch input
-  processSwitches();
+  check = false;
+  processSwitches(check);
 
   // bottom of loop()
 }
 
-void processSwitches(void) {
-  bool isSuccess = true;
-  for (byte i = 0; i < NUM_SWITCHES; i++) {
-    moduleSwitch[i].update();
-    if (moduleSwitch[i].changed()) {      
-      byte nv = i+1;
-      byte nvval = modconfig.readNV(nv);
-      bool state;
-
-      //DEBUG_PRINT(F("> Button ") << i << F(" state change detected"));
-      //DEBUG_PRINT(F(" NV Value = ") << nvval);
-
-      switch (nvval) {
-        case 0:
-          // ON and OFF
-          state = (moduleSwitch[i].fell());
-          //DEBUG_PRINT(F("> Button ") << i << (moduleSwitch[i].fell() ? F(" pressed, send state: ") : F(" released, send state: ")) << state);
-          epService.sendEvent(state, i);
-          break;
-
-        case 1:
-          // Only ON
-          if (moduleSwitch[i].fell()) {
-            state = true;
-            //DEBUG_PRINT(F("> Button ") << i << F(" pressed, send state: ") << state);
-            epService.sendEvent(state, i);
-          }
-          break;
-
-        case 2:
-          // Only OFF
-          if (moduleSwitch[i].fell()) {
-            state = false;
-            //DEBUG_PRINT(F("> Button ") << i << F(" pressed, send state: ") << state);
-            epService.sendEvent(state, i);
-          }
-          break;
-
-        case 3:
-          // Toggle button
-          if (moduleSwitch[i].fell()) {
-            switchState[i] = !switchState[i];
-            state = (switchState[i]);
-            //DEBUG_PRINT(F("> Button ") << i << (moduleSwitch[i].fell() ? F(" pressed, send state: ") : F(" released, send state: ")) << state);
+byte checkProduced(void)
+{
+  check = true;
+  byte pIndex = processSwitches(check);
+  return pIndex;
+}
+
+byte processSwitches(bool check) {
```

Comment:
This function does two things. 1) Check which switch has been changed and returns an index; 2) checks which swich has been changed (same so far) and sends an event. Note that for 2) it doesn't return anything eventhough it has a return type. 
Looks like these two things should be different functions. If you need to do one or another, please create a third function that calls either of the two functions. Keep the functions small.

Reply:
Oh so simple!  You wouldn't believe the trouble I had thinking this one through!